Automattic/wp-calypso
# Media: show a refresh button for external media

While viewing Google Photos the user may externally add new photos (i.e. they take a new photo and it gets uploaded to GPhotos). This PR adds a refresh button to allow the user to force the media library to refresh the list, pulling in new media.

<img width="252" alt="refresh" src="https://user-images.githubusercontent.com/1277682/28774987-b88687c0-75e7-11e7-9ac8-460f460cd917.png">

The button is disabled while the library is fetching data. Due to the way the infinite list works, this disabled state can be toggled on and off quickly, causing the button to flicker. To work around this the component debounces it:

- If the library is fetching data then we instantly disable the button
- If the library is not fetching data we debounce enabling the button

Note that auto-refreshing could be an option for the future, but is trickier to get right.

### Testing

- Open up Google Photos in the media library and note the refresh button is now present
- Verify that the button is disabled while the media library is being populated (note that it loads 3 pages of media)
- When the button is enabled click it to cause the library to empty and be refreshed
